pkgs/development/tools/misc/hydra/default.nix

```diff
@@ -79,16 +79,17 @@ in releaseTools.nixBuild rec {
       guile # optional, for Guile + Guix support
       perlDeps perl nixUnstable
       postgresql # for running the tests
-      (lib.overrideDerivation (aws-sdk-cpp.override {
+      ((aws-sdk-cpp.override {
         apis = ["s3"];
         customMemoryManagement = false;
-      }) (attrs: {
+      }).overrideAttrs (attrs: {
         src = fetchFromGitHub {
           owner = "edolstra";
           repo = "aws-sdk-cpp";
           rev = "local";
           sha256 = "1vhgsxkhpai9a7dk38q4r239l6dsz2jvl8hii24c194lsga3g84h";
         };
+        patches = attrs.patches or [] ++ [ ./aws-sdk-cpp-gcc7.patch ];
```

Comment:
Wouldn't it be better to apply this patch in the `aws-sdk-cpp` expression?

Reply:
No.  It's a different source and doesn't need it.

Reply:
I would be better not to use yet another version of the package, but there was probably a good reason for that.

Reply:
No, it's no longer needed. In fact Hydra doesn't need aws-sdk-cpp at all anymore (https://github.com/NixOS/hydra/commit/3ec73c93e2cdf58750ab183746627892ced55c4f).

Reply:
Updated the PR to remove it.  I didn't realize AWS is now handled only by nix directly.